executor/common.h

```diff
@@ -371,8 +371,11 @@ static void csum_inet_update(struct csum_inet* csum, const uint8* data, size_t l
 	for (i = 0; i < length - 1; i += 2)
 		csum->acc += *(uint16*)&data[i];
 
-	if (length & 1)
-		csum->acc += (uint16)data[length - 1];
+	if (length & 1) {
+		uint16 tmp = 0;
+		*(uint8*)&tmp = data[length - 1];
```

Comment:
Such cast smells like an undefined behavior. Can't we use something like `be16toh(data[length - 1])`?

Reply:
When calculating
the checksum of odd number of bytes, the last byte must be
interpreted as LSB on little-endian architectures and
as MSB on big-endian ones in a 16-bit half-word.

Reply:
Yes, but this line looks like an undefined behavior. Is it legal to stomp a random byte of a variable?

Reply:
not sure, would be interesting to find a reference in the C++ standard, in the mean time i changed it :)

Reply:
btw, i saw such casts in a lot of C code out there in the wild, if it is undefined behavior then we would be not alone :)

Reply:
Yes, there are lots of buggy code out there. You won't surprise me with this :)

Not the standard, but looks legit:
https://adriann.github.io/undefined_behavior.html

Reply:
Using `be16toh()` here is really confusing, calculating the checksum should be byte order independent. I think we should do this as shown in section 4 of the [RFC](https://tools.ietf.org/html/rfc1071). Casting pointers to `char *` should be valid.

Reply:
i think rfc is not entirely correct at this place. it works only for LE.

Reply:
an example from busybox: https://git.buildroot.org/busybox/tree/libbb/inet_cksum.c

Reply:
OK, I see. Let's do this then:

```
csum->acc += le16toh((uint16)data[length - 1]);
```

And use `le16toh()` in tests too.